site/process_sdk_docs.py

```diff
@@ -0,0 +1,296 @@
+#!/usr/bin/env python
+
+# Copyright (C) 2022 CVAT.ai Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import argparse
+import os
+import os.path as osp
+import re
+import shutil
+import sys
+import textwrap
+from glob import iglob
+from typing import Callable, List
+
+from inflection import underscore
+
+
+class Processor:
+    _reference_files: List[str]
+
+    def __init__(self, *, input_dir: str, site_root: str) -> None:
+        self._input_dir = input_dir
+        self._site_root = site_root
+
+        self._content_dir = osp.join(self._site_root, "content")
+        self._sdk_reference_dir = osp.join(
+            self._content_dir, "en/docs/api_sdk/sdk/reference"
+        )
+        self._templates_dir = osp.join(self._site_root, "templates")
+
+    @staticmethod
+    def _copy_files(src_dir: str, glob_pattern: str, dst_dir: str) -> List[str]:
+        copied_files = []
+
+        for src_path in iglob(osp.join(src_dir, glob_pattern), recursive=True):
+            src_filename = osp.relpath(src_path, src_dir)
+            dst_path = osp.join(dst_dir, src_filename)
+            # assume dst dir exists
+            shutil.copy(src_path, dst_path, follow_symlinks=True)
+
+            copied_files.append(dst_path)
+
+        return copied_files
+
+    def _copy_pages(self):
+        self._reference_files = self._copy_files(
+            self._input_dir, "*/**/*.md", self._sdk_reference_dir
+        )
+
+    def _add_page_headers(self):
+        """
+        Adds headers required by hugo to docs pages
+        """
+
+        HEADER_SEPARATOR = "---"
+
+        for p in self._reference_files:
+            with open(p) as f:
+                contents = f.read()
+
+            assert not contents.startswith(HEADER_SEPARATOR), p
+
+            lines = contents.splitlines()
+
+            assert lines[0].startswith("#")
+            classname = lines[0][1:].strip()
+
+            header = textwrap.dedent(
+                """\
+                %(header_separator)s
+                title: '%(classname)s class reference'
+                linkTitle: '%(classname)s'
+                weight: 10
+                description: ''
+                %(header_separator)s
+            """
+                % {"header_separator": HEADER_SEPARATOR, "classname": classname}
+            )
+
+            contents = header + "\n".join(lines[1:])
+
+            with open(p, "w") as f:
+                f.write(contents)
+
+    def _extract_apis_summary(self, readme_path: str) -> str:
+        with open(readme_path) as f:
+            readme_contents = f.read()
+
+        apis_summary = re.findall(
+            r"## Available API Endpoints(.*)## Available Models",
+            readme_contents,
+            flags=re.DOTALL,
+        )
+        apis_summary = apis_summary[0].strip()
```

Comment:
Why use `findall` if you only use the first result?

Reply:
No specific reason, changed to search.